src/Storefront/Resources/app/storefront/src/component-system/README.md

```diff
@@ -0,0 +1,361 @@
+
+```javascript
+// BuyButton.js
+
+class BuyButton extends ShopwareComponent {
+
+    // ...
+
+    onFormSubmit(event) {
+        event.preventDefault();
+
+        let requestUrl = this.el.getAttribute('action');
+        let formData = window.Shopware.serializeForm(this.el);
+
+        [ requestUrl, formData ] = window.Shopware.emitInterception('BuyButton:PreSubmit', requestUrl, formData);
+
+        window.Shopware.emit('BuyButton:Submit', requestUrl, formData);
+
+        window.Shopware.callPluginMethod('OffCanvasCart', 'openOffCanvas', requestUrl, formData);
+    }
+}
+```
+
+You can see that the event `BuyButton:PreSubmit` offers the opportunity to manipulate the `formData` before it gets sent. From any other script you can intercept this event and work with the parameters from the event.
+
+```javascript
+// Intercept the buy button event
+window.Shopware.intercept('BuyButton:PreSubmit', (requestUrl, formData) => {
+    
+    formData.append('foo', 'bar');
+
+    return [ requestUrl, formData ];
+});
+```
+
+Don't forget to return the passed parameters in the same order, so the component logic can work with it. 
+
+There can be multiple subscribers to a single event. They will all be executed in the order as they are registered. You can change the order by passing a priority parameter as an optional third option, when registering an event. By default all subscribers have the priority `0`. The higher the priority the earlier the subscriber is called in the chain. Also negative values are possible to move a subscriber further down the chain.
+
+```javascript
+// Another interceptor to the buy button event
+window.Shopware.intercept('BuyButton:PreSubmit', (requestUrl, formData) => {
+    
+    formData.delete('foo');
+    formData.append('bar', 'baz')
+
+    return [ requestUrl, formData ];
+}, -10);
+```
+
+### Method Calling
+
+Besides the event system you can also access other component instances directly, or call methods for all active instances of a component.
+
+```javascript
+// Call a method on all instances of a component
+Shopware.callMethod('MyComponent', 'refresh');
+
+// Get all instances of a component
+const instances = Shopware.getComponentInstances('MyComponent');
+
+// Get a specific instance by element
+const instance = Shopware.getComponentInstanceByElement('MyComponent', element);
+```
+
+
+
+### Mutation Observation
+
+Components can observe DOM and attribute changes on their elements and children. The component base class offers an optional mutation observer that can be started separately if needed.
+
+You can call `initializeObserver()` in your component to start the observer and pass the desired observer configuration. If you want to use this, there are two additional lifecycle methods available to react to content and attribute changes.
```

Comment:
Will it automatically re-init by default if the mutation observer sees an additional selector from a JS plugin? Maybe this would be also worth noticing here.

Reply:
It is a separate thing and has nothing to do with the automatic initialization. The auto-init will always work and is controlled by the Shopware singleton. The observer for the component is separate and optional.